Infrastructure/Services/TrackingNumberGenerator.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Infrastructure.Interfaces;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Services
+{
+    public class TrackingNumberGenerator : ITrackingNumberGenerator
+    {
+        private readonly ILogger<TrackingNumberGenerator> _logger;
+
+        public TrackingNumberGenerator(ILogger<TrackingNumberGenerator> logger)
+        {
+            _logger = logger;
+        }
+
+        public string GenerateNumber()
+        {
+            Guid g = Guid.NewGuid();
+            var trackingNumber = g.ToString().Remove(0, 24);
+            _logger.LogInformation("Executed {0}->Generated new trackingNumber", this.GetType().Name);
```

Comment:
Method name should be received from code instead of hardcoded string. i.e. If I were to change method's name, I should not edit this line...